subsys/bluetooth/host/hci_core.c

```diff
@@ -3641,6 +3641,50 @@ int bt_enable(bt_ready_cb_t cb)
 	return 0;
 }
 
+int bt_disable(void)
+{
+	int err;
+
+	if (!bt_dev.drv) {
+		BT_ERR("No HCI driver registered");
+		return -ENODEV;
+	}
+
+	atomic_clear_bit(bt_dev.flags, BT_DEV_ENABLE);
+
+	if (atomic_test_and_set_bit(bt_dev.flags, BT_DEV_DISABLE)) {
+		return -EALREADY;
+	}
+
+	if (!bt_dev.drv->close) {
+		return -ENOTSUP;
+	}
```

Comment:
You need to have this check before making any state modifications, i.e. you need to move it before clearing `BT_DEV_ENABLE`.

Reply:
Right, fixed.